### PR TITLE
Add giveth gnosis decoders and balance query through givPOW underlying asset

### LINKED ISSUE #9005: Track GIV tokens staked in GIVpower program
## Abstract

I would like to see the amount & value of GIV I have staked within Giveth's staking program. https://giveth.io/givfarm

## Motivation

I'm a Giveth maxi so it makes up a notable amount of my portfolio - I would like to have it considered when looking at my assets on rotki. 

## Specification

Staking is deployed on 3 chains atm - Gnosis, Optimism and Polygon zkEVM. Docs regarding GIVpower staking can be found here: 
https://docs.giveth.io/giveconomy/givpower

coingecko listing is here: https://www.coingecko.com/en/coins/giveth

### How to get balances

**Gnosis** - users wrap/stake their GIV tokens into gGIV (Giveth Governance) was used in a complex aragon DAO structure wrapped by 1hive Gardens DAO framework. 
TL;DR - get user balance from this function -> https://gnosis.blockscout.com/address/0xfFBAbEb49be77E5254333d5fdfF72920B989425f?tab=read_write_contract#0x70a08231

**Optimism** 
not so complex, balance can be found by querying here -> https://optimistic.etherscan.io/address/0x301c739cf6bfb6b47a74878bdeb13f92f13ae5e7#readProxyContract#F12

**Polygon zk-evm**
same deal here 
https://zkevm.polygonscan.com/address/0xc790f82bf6f8709aa4a56dc11afad7af7c2a9867#readProxyContract#F12

### PR DESCRIPTION
**Pull request by LefterisJP**
Closes #9005 

Needs https://github.com/rotki/assets/pull/213  (I think)

### CODE REVIEW

File: rotkehlchen/chain/evm/decoding/giveth/decoder.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            events_list=context.decoded_events,
+        )
+        return DecodingOutput(event=lock_event)
+
+    def decode_claim(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] != SIMPLE_CLAIM:
+            return DEFAULT_DECODING_OUTPUT
+
+        if not self.base.is_tracked(user := bytes_to_address(context.tx_log.topics[1])):
+            return DEFAULT_DECODING_OUTPUT
```
Comment (yabirgb) [inline rotkehlchen/chain/evm/decoding/giveth/decoder.py:105]: can't this two be in a or statement?

File: rotkehlchen/chain/evm/decoding/giveth/decoder.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+        amount = token_normalized_value_decimals(
+            token_amount=int.from_bytes(context.tx_log.data[:32]),
+            token_decimals=DEFAULT_TOKEN_DECIMALS,  # GIV has 18 decimals
+        )
+        for event in context.decoded_events:
+            if (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.asset.identifier == self.giv_token_id and
```
Comment (yabirgb) [inline rotkehlchen/chain/evm/decoding/giveth/decoder.py:114]: comparing the asset will also compare the identifier right? but not really an important thing

File: rotkehlchen/chain/gnosis/modules/giveth/decoder.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            amount=amount,
+            location_label=user,
+            asset=Asset(self.pow_token_id),
+            address=ZERO_ADDRESS,
+            to_event_subtype=HistoryEventSubType.RECEIVE_WRAPPED,
+            to_location_label=user,
+            to_address=context.tx_log.address,
+            to_notes='Receive {amount} POW after depositing GIV',  # to be filled by the action item  # noqa: E501
+            to_counterparty=CPT_GIVETH,
+            paired_events_data=([out_event], True),  # make sure order is respected
```
Comment (yabirgb) [inline rotkehlchen/chain/gnosis/modules/giveth/decoder.py:105]: what happens if out_event is None here?

File: rotkehlchen/chain/evm/constants.py [hunk 4]
```diff
@@ -40,3 +40,5 @@
 
 # uniswap like merkle distributor claimed signature
 MERKLE_CLAIM: Final = b'N\xc9\x0e\x96U\x19\xd9&\x81&tg\xf7u\xad\xa5\xbd!J\xa9,\r\xc9=\x90\xa5\xe8\x80\xce\x9e\xd0&'  # noqa: E501
+# simple claim for address and amount used by multiple protocols
+SIMPLE_CLAIM: Final = b'G\xce\xe9|\xb7\xac\xd7\x17\xb3\xc0\xaa\x145\xd0\x04\xcd[<\x8cW\xd7\r\xbc\xebNDX\xbb\xd6\x0e9\xd4'  # noqa: E501
```
[the remark this answers is not in the document]
Comment (LefterisJP) [reply rotkehlchen/chain/evm/constants.py:44]: this was how it was before. I did not want to do more changes than I already did in one PR.

File: rotkehlchen/chain/evm/decoding/giveth/constants.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+from typing import Final
+
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+
+CPT_GIVETH: Final = 'giveth'
+CPT_DETAILS_GIVETH: Final = CounterpartyDetails(
+    identifier=CPT_GIVETH,
+    label='Giveth',
+    image='giveth.jpg',
```
[the remark this answers is not in the document]
Comment (LefterisJP) [reply rotkehlchen/chain/evm/decoding/giveth/constants.py:9]: This is already there. I did not add it in this PR

File: rotkehlchen/chain/evm/decoding/giveth/decoder.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class GivethDecoderBase(DecoderInterface, ABC):
+
+    def __init__(  # pylint: disable=super-init-not-called
+            self,
```
[the remark this answers is not in the document]
Comment (LefterisJP) [reply rotkehlchen/chain/evm/decoding/giveth/decoder.py:36]: yes indeed it's not required

File: rotkehlchen/chain/evm/decoding/giveth/decoder.py [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+                receive_event = event
+                break
+
+        else:
+            log.error(f'Could not find the GivPoW token transfer after locking GIV for {context.transaction}')  # noqa: E501
+
+        maybe_reshuffle_events(
+            ordered_events=[lock_event, receive_event],
+            events_list=context.decoded_events,
+        )
```
[the remark this answers is not in the document]
Comment (LefterisJP) [reply rotkehlchen/chain/evm/decoding/giveth/decoder.py:97]: there is already a check inside the function for this. This is why it's called `maybe_reshufle_events`. Doing it here would save us an iteration maybe in the case it's None. Not worth it.

File: rotkehlchen/chain/evm/decoding/giveth/decoder.py (hunk 1 above)
Comment (LefterisJP) [reply rotkehlchen/chain/evm/decoding/giveth/decoder.py:105]: yes. Let me add them

File: rotkehlchen/chain/evm/decoding/giveth/decoder.py (hunk 2 above)
Comment (LefterisJP) [reply rotkehlchen/chain/evm/decoding/giveth/decoder.py:114]: we only have identifiers here. This save us unecessary asset instantiations.

Comment (yabirgb) [reply rotkehlchen/chain/evm/decoding/giveth/decoder.py:114]: I meant that you can do 

```
event.asset == self.giv_token_id
```

due to the overload in the eq operator

File: rotkehlchen/chain/gnosis/modules/giveth/decoder.py (hunk 3 above)
Comment (LefterisJP) [reply rotkehlchen/chain/gnosis/modules/giveth/decoder.py:105]: It can't be None here

**Pull request merged**